control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -174,24 +174,38 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 			Status: metav1.ConditionUnknown,
 			Reason: "EtcdStatusUnknown",
 		}
-		etcdCluster := manifests.EtcdCluster(hostedControlPlane.Namespace)
-		if err := r.Get(ctx, types.NamespacedName{Namespace: etcdCluster.Namespace, Name: etcdCluster.Name}, etcdCluster); err != nil {
-			if apierrors.IsNotFound(err) {
-				newCondition = metav1.Condition{
-					Type:   string(hyperv1.EtcdAvailable),
-					Status: metav1.ConditionFalse,
-					Reason: "EtcdClusterNotFound",
+		switch hostedControlPlane.Spec.Etcd.ManagementType {
+		case hyperv1.Managed:
+			r.Log.Info("Reconciling etcd cluster status for managed strategy")
+			etcdCluster := manifests.EtcdCluster(hostedControlPlane.Namespace)
+			if err := r.Get(ctx, types.NamespacedName{Namespace: etcdCluster.Namespace, Name: etcdCluster.Name}, etcdCluster); err != nil {
+				if apierrors.IsNotFound(err) {
+					newCondition = metav1.Condition{
+						Type:   string(hyperv1.EtcdAvailable),
+						Status: metav1.ConditionFalse,
+						Reason: "EtcdClusterNotFound",
+					}
+				} else {
+					return ctrl.Result{}, fmt.Errorf("failed to fetch etcd cluster %s/%s: %w", etcdCluster.Namespace, etcdCluster.Name, err)
 				}
 			} else {
-				return ctrl.Result{}, fmt.Errorf("failed to fetch etcd cluster %s/%s: %w", etcdCluster.Namespace, etcdCluster.Name, err)
+				r.Log.Info("Computing proper etcd cluster status based on current state of etcd cluster")
+				cond, err := etcd.ComputeEtcdClusterStatus(ctx, r.Client, etcdCluster)
+				if err != nil {
+					return ctrl.Result{}, fmt.Errorf("failed to compute etcd cluster status: %w", err)
+				}
+				newCondition = cond
 			}
-		} else {
-			cond, err := etcd.ComputeEtcdClusterStatus(ctx, r.Client, etcdCluster)
-			if err != nil {
-				return ctrl.Result{}, fmt.Errorf("failed to compute etcd cluster status: %w", err)
+		case hyperv1.Unmanaged:
+			r.Log.Info("Assuming Etcd cluster is running in unmanaged etcd strategy")
+			newCondition = metav1.Condition{
+				Type:    string(hyperv1.EtcdAvailable),
+				Status:  metav1.ConditionTrue,
+				Reason:  etcd.EtcdReasonRunning,
+				Message: "Etcd cluster is assumed to be running in unmanaged state",
 			}
-			newCondition = cond
 		}
+		r.Log.Info("Updating hosted control plane status")
```

Comment:
this log line seems confusing. It would only be printed for the etcd status. I think it makes sense to log when the hcp status is actually updated in ~line 354.

Reply:
removed